### PR TITLE
Add notes to more accurately describe misaka library extensions

### PR DESCRIPTION
**Pull request by mgc8**
Added note to the "math" extension to clarify that it doesn't actually render any math, just marks it as ignored by the parser, so that it can be processed with external JavaScript libraries separately. Added notes to the "highlight", "tables" and "underline" extensions with the requirements for each to actually work when used.

## Checklist
- [x] All new and existing **tests are passing**
- [x] (If adding features:) I have added tests to cover my changes
- [x] (If docs changes needed:) I have updated the **documentation** accordingly.
- [ ] I have added an entry to `CHANGES.rst` because this is a user-facing change or an important bugfix
- [ ] I have written **proper commit message(s)**
      

## What changes does this Pull Request introduce?

Documentation changes only -- clarifications on the functionality of ``misaka`` extensions.

## Why is this necessary?

The existing documentation was incomplete.

### CODE REVIEW

**Review by ix5 — CHANGES_REQUESTED** [review]
Nice, thank you!

File: docs/docs/reference/markdown-config.rst [hunk 1]
```diff
@@ -127,8 +127,14 @@ The following **additional options** are available:
...
 | Ignore indented code blocks                              | ``disable-indented-code`` |
 +----------------------------------------------------------+---------------------------+
-| Parse inline LaTeX-style math blocks, such as            | ``math``                  |
-| inline ``$equations$`` or display ``$$equations$$``      |                           |
+| Ignore inline LaTeX-style math blocks, such as           | ``math``                  |
+| inline ``$equations$`` or display ``$$equations$$``,     |                           |
+| allowing them to be processed separately with a          |                           |
+| JavaScript library.                                      |                           |
+| **Note:** This extension will *not* render equations     |                           |
+| or any form of math, it just marks them to be ignored!   |                           |
```
[the remark this answers is not in the document]
Comment (mgc8) [reply docs/docs/reference/markdown-config.rst]: Fixed in latest commit.

**Review by ix5 — CHANGES_REQUESTED** [review]
Thank you for applying the feedback.

Per `git` conventions and in order for this project to have a meaningful and well-searchable `git` history, we ask contributors to keep commit title length to 50 characters and the body to a maximum of 80.
You can look at `git log -- docs/` or browse the [history of the docs/ folder](https://github.com/isso-comments/isso/commits/master/docs) to see examples of how commits _should_ be formatted.

We also prefer you squash superfluous commits (this change should only consist of 1 logical commit since it only touches 1 file). You can do so with `git rebase -i master` and change `pick` to `squash`. Otherwise our history would be littered with multiple rounds of PR feedback, making it difficult to track down with `git blame` through which commit (and with which intentions/justification) changes to files were made.

**Comment by mgc8** [comment]
Sorry for the wrong format, I've made the changes locally to squash the commits and re-phrase, then pushed it over the existing PR. I hope it's fine now, please have a look and let me know if other changes are necessary.

**Comment by ix5** [comment]
Thank you.

One last thing, please fix the commit title length. E.g. `docs: markdown-config: Clarify misaka config` would be succinct enough.

We'd want to avoid this:
Screenshot from 2024-04-29 23-14-31

**Comment by mgc8** [comment]
Thank you for taking the time on the reviews as well.

> One last thing, please fix the commit title length. E.g. `docs: markdown-config: Clarify misaka config` would be succinct enough.

Yeah, the funny thing is that I noticed that, and actually did a "reword" when I squashed the commits and in my local logs it appeared fine; but for some reason it didn't update when I pushed things to GitHub :-/ Could be that my local copy became a bit confused with all the re-basing happening from the other commits that come in, so I deleted everything, re-pulled and re-based again, I hope it's fine now?

**Review by ix5 — APPROVED** [review]

**Comment by ix5** [comment]
Merged, thanks!

**Pull request closed without merging**